sklearn/ensemble/forest.py

```diff
@@ -417,6 +418,8 @@ def _set_oob_score(self, X, y):
         self.oob_score_ = oob_score / self.n_outputs_
 
     def _validate_y_class_weight(self, y):
+        assert_non_regression_targets(y)
```

Comment:
you could do ``check_non_regression_target``? On second though, as it raises an exception, assert seems ok.